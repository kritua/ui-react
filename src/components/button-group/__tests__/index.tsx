import React from 'react';
import renderer from 'react-test-renderer';

import { ButtonGroup } from './../index';

it('Desktop Row', () => {
    const tree = renderer.create(<ButtonGroup presetStyle="desktop" direction="row" />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('Desktop Column', () => {
    const tree = renderer.create(<ButtonGroup presetStyle="desktop" direction="column" />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('Mobile Row', () => {
    const tree = renderer.create(<ButtonGroup presetStyle="mobile" direction="row" />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('Mobile Column', () => {
    const tree = renderer.create(<ButtonGroup presetStyle="mobile" direction="column" />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('tagName', () => {
    const tree = renderer.create(<ButtonGroup tagName="span" children="MOCK" />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('tagName empty string (storybook)', () => {
    const tree = renderer.create(<ButtonGroup tagName="" children="MOCK" />).toJSON();

    expect(tree).toMatchSnapshot();
});
