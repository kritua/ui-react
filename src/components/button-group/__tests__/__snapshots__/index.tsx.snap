// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop Column 1`] = `
<div
  className="button-group button-group_column button-group_desktop"
/>
`;

exports[`Desktop Row 1`] = `
<div
  className="button-group button-group_row button-group_desktop"
/>
`;

exports[`Mobile Column 1`] = `
<div
  className="button-group button-group_column button-group_mobile"
/>
`;

exports[`Mobile Row 1`] = `
<div
  className="button-group button-group_row button-group_mobile"
/>
`;

exports[`tagName 1`] = `
<span
  className="button-group button-group_column button-group_desktop"
>
  MOCK
</span>
`;

exports[`tagName empty string (storybook) 1`] = `
<div
  className="button-group button-group_column button-group_desktop"
>
  MOCK
</div>
`;
